From 1a54b008068f9e6c99bd968f38392ca88f93c2f0 Mon Sep 17 00:00:00 2001
From: Peter Harliman Liem <peter.harliman.liem@intel.com>
Date: Tue, 18 Sep 2018 16:13:12 +0800
Subject: [PATCH] DRVLIB_SW-930 - Update gphy pin strapping register

Update resistor calibration (RCAL) and RC measurement (RC_COUNT)
as recommended by verification & gphy team.
---
 drivers/net/ethernet/lantiq/xrx500_phy_fw.c | 15 ++++++++++-----
 1 file changed, 10 insertions(+), 5 deletions(-)

diff --git a/drivers/net/ethernet/lantiq/xrx500_phy_fw.c b/drivers/net/ethernet/lantiq/xrx500_phy_fw.c
index 6ef08b0a2047..510e18479c26 100644
--- a/drivers/net/ethernet/lantiq/xrx500_phy_fw.c
+++ b/drivers/net/ethernet/lantiq/xrx500_phy_fw.c
@@ -78,8 +78,6 @@ static u32 xrx500_gphy[] = {
 #define PRX300_GPHY_FCR 0x800
 #define PRX300_GPHY0_GPS0 0x804
 #define PRX300_GPHY0_GPS1 0x808
-#define PRX300_GPHY0_GPS0_LO 0x3f004000
-#define PRX300_GPHY0_GPS0_HI 0
 /* GPHY CDB */
 #define PRX300_GPHY_CDB_PDI_PLL_CFG0 0x0
 #define PRX300_GPHY_CDB_PDI_PLL_CFG2 0x8
@@ -156,6 +154,7 @@ static int xrx500_dt_parse(struct xway_gphy_data *priv)
 static int prx300_gphy_boot(struct xway_gphy_data *priv)
 {
 	struct prx300_reset_control *rst = &priv->rst.prx300;
+	u32 pin_strap_lo, pin_strap_hi;
 
 	/* set LAN interface to GPHY */
 	regmap_update_bits(priv->syscfg, PRX300_IFMUX_CFG, PRX300_LAN_MUX_MASK,
@@ -173,13 +172,19 @@ static int prx300_gphy_boot(struct xway_gphy_data *priv)
 	/* release CDB reset */
 	reset_control_deassert(rst->gphy_cdb);
 
-	/* GPHY FW address and pin strapping */
+	/* GPHY FW address */
 	regmap_update_bits(priv->cgu_syscfg, PRX300_GPHY_FCR, ~0,
 			   priv->dma_addr);
+
+	pin_strap_lo = 0x4000; /* base freq deviation */
+	pin_strap_lo |= 0x1f << 24; /* MDIO address */
+	pin_strap_lo |= 0x1 << 29; /* interrupt polarity */
+	pin_strap_hi = 0x8; /* RCAL */
+	pin_strap_hi |= 0x10 << 4; /* RC count */
 	regmap_update_bits(priv->cgu_syscfg, PRX300_GPHY0_GPS0, ~0,
-			   PRX300_GPHY0_GPS0_LO);
+			   pin_strap_lo);
 	regmap_update_bits(priv->cgu_syscfg, PRX300_GPHY0_GPS1, ~0,
-			   PRX300_GPHY0_GPS0_HI);
+			   pin_strap_hi);
 
 	/* release GPHY reset */
 	reset_control_deassert(rst->gphy);
